### PR TITLE
feat(sso): enforce SSO on self-hosted instances

### LINKED ISSUE #7535: Ability to restrict users to SSO login only
## Is your feature request related to a problem?

We manage around 20 3rd party tools that our internal users can log into using G Suite (part of your enterprise / scale offering). When a user leaves, we need to disable their account. Currently we need to do this in G Suite and then any of the 3rd party tools where password login is still possible.

## Describe the solution you'd like

When a user leaves, we disable their account in G Suite and their account in PostHog becomes disabled too. Essentially, the ability to restrict specific users in PostHog to only be able to login via SSO (defaulting to this setting).

## Describe alternatives you've considered

Manually removing their account, and being the enemy of our IT manager.

### PR DESCRIPTION
**Pull request by paolodamico**
## Problem
Closes #7535 (context there).

## Changes
Instance admins can now configure an `SSO_ENFORCEMENT` parameter via environment variables (with value `saml`, `google-oauth2`, `github` or `gitlab`) so that only SSO login is allowed on the instance. SSO must be properly configured **and appropriate valid license must be present and active.**

⚠️ `SAML_ENFORCED` parameter is now deprecated. 1.34.0 will still support this parameter (as long as `SSO_ENFORCEMENT` is not set) but a deprecation warning will be logged. 

In addition to the above, we now validate the instance has a valid license to use Google or SAML respectively to enable the actual login flow with those providers (previously it was a client-side validation only). If a license expires, and enforcement was active, it will be terminated and users will be able to log in with regular password (presumably to fix their license situation).

## How did you test this code?
Django functional tests.

### CODE REVIEW

File: ee/settings.py [hunk 1]
```diff
@@ -45,17 +34,6 @@
 SOCIAL_AUTH_SAML_SUPPORT_CONTACT = SOCIAL_AUTH_SAML_TECHNICAL_CONTACT
 
 
-def getenv(key, default=""):
```
Comment (paolodamico) [inline ee/settings.py:48]: Remove this duplicate function in favor of the more complete `get_from_env`

File: ee/settings.py [hunk 2]
```diff
@@ -19,20 +19,9 @@
 HOOK_FINDER = "ee.models.hook.find_and_fire_hook"
 HOOK_DELIVERER = "ee.models.hook.deliver_hook_wrapper"
 
-# Social auth
```
Comment (paolodamico) [inline ee/settings.py:22]: These lines (22 - 32) were copied verbatim to L61 - onwards to account for needed variable access at the bottom.

**Comment by paolodamico** [comment]
Ping @mariusandra

**Review by mariusandra — APPROVED** [review]
Looks good.

**Pull request merged**